components/external_storage/src/lib.rs

```diff
@@ -85,17 +116,25 @@ pub trait ExternalStorage: 'static + Send + Sync {
         &self,
         storage_name: &str,
         restore_name: std::path::PathBuf,
-        compressed_range: Option<(u64, u64)>,
         expected_length: u64,
-        expected_sha256: Option<Vec<u8>>,
         speed_limiter: &Limiter,
         file_crypter: Option<FileEncryptionInfo>,
```

Comment:
Can we move `file_crypter` into `Options` too? So the `Options` fields contains all optional arguments of `restore`.

Reply:
done